lms/djangoapps/instructor/views/api.py

```diff
@@ -1038,32 +1047,52 @@ def get_problem_responses(request, course_id):
         "task_id": "4e49522f-31d9-431a-9cff-dd2a2bf4c85a"
     }
     ```
-
-    Responds with BadRequest if any of the provided problem locations are faulty.
     """
-    course_key = CourseKey.from_string(course_id)
-    # A comma-separated list of problem locations
-    # The name of the POST parameter is `problem_location` (not pluralised) in
-    # order to preserve backwards compatibility with existing third-party
-    # scripts.
-    problem_locations = request.POST.get('problem_location', '')
-    # A comma-separated list of block types
-    problem_types_filter = request.POST.get('problem_types_filter', '')
-    report_type = _('problem responses')
 
-    try:
-        for problem_location in problem_locations.split(','):
-            problem_key = UsageKey.from_string(problem_location).map_into_course(course_key)
-    except InvalidKeyError:
-        log.exception("ProblemResponsesError: Location: %s", problem_locations)
-        return JsonResponseBadRequest(_("Could not find problem with this location."))
+    # The inclusion of SessionAuthentication here should require that a valid CSRF token
+    # is submitted on any session based requests
+    authentication_classes = (JwtAuthentication, OAuth2AuthenticationAllowInactiveUser, SessionAuthentication,)
+    permission_classes = (IsAuthenticated, IsCourseStaffInstructor,)
```

Comment:
The current approach is to add the `@view_auth_classes()` decorator.

Reply:
I had to revert the changes to `@view_auth_classes` after looking through the code. These views have specific permissions classes (in this case, they require `IsCourseStaffInstructor`), and `@view_auth_classes` has no mechanism to add additional permissions classes. I looked through all the code and could not find any instance of `@view_auth_classes` being used, and additional permissions classes also being included on a class based view. Conversely, I found circumstances that, within the same file, `@view_auth_classes` was being used on view classes that did not have a `permission_classes` attribute, and was not being used on the view classes that did have that attribute. Effectively, I think they are mutually exclusive, and in this case, `permission_classes` is required.